pkg/distributor/receiver/shim_test.go

```diff
@@ -0,0 +1,202 @@
+package receiver
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/logging"
+	"go.opentelemetry.io/collector/model/otlpgrpc"
+	"go.opentelemetry.io/collector/model/pdata"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/grafana/loki/pkg/logproto"
+)
+
+func TestParseEntry(t *testing.T) {
+	pLog := pdata.NewLogRecord()
+	pLog.SetName("testName")
+	pLog.SetFlags(31)
+	pLog.SetSeverityNumber(1)
+	pLog.SetSeverityText("WARN LEVEL")
+	pLog.SetSpanID(pdata.NewSpanID([8]byte{1, 2}))
+	pLog.SetTraceID(pdata.NewTraceID([16]byte{1, 2, 3, 4}))
+
+	now := time.Now()
+	pLog.SetTimestamp(pdata.NewTimestampFromTime(now))
+
+	entry, err := parseEntry(pLog, "json")
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
I think we should use `require.NoError(t, err)` for checking.

Reply:
done,thanks